### ISSUE TITLE
fasl-write problems with 7.4cs while installing rosette

### ISSUE THREAD

**Issue by pmatos** [racket/racket#2789]
Installing `rosette` with `raco pkg install --auto rosette` in 7.4cs causes:
```
fasl-write: invalid fasl object #<procedure:cons>
  compilation context...:
   /root/.racket/7.3.0.900/pkgs/rosette/rosette/lib/synthax/form.rkt
   /root/.racket/7.3.0.900/pkgs/rosette/rosette/lib/synthax.rkt
  context...:
   condition->exn
   do-raise
   dynamic-wind
   c-faslcode
   c-faslcode
   c-faslcode
   c-faslcode
   start
   dynamic-wind
   dynamic-wind
   compile-to-bytevector
   thunk
   compile-module-linklet57
   for-loop
   compile-forms33
   compile-module13
```

I am on holidays so I can't look into this in-depth at the moment, but I will if nobody else does once I am back.

**Comment by pmatos** [comment]
Installation finishes with:
```
raco setup: error: during making for <pkgs>/rosette/rosette/lib
raco setup:   fasl-write: invalid fasl object #<procedure:cons>
raco setup:     compiling: <pkgs>/rosette/rosette/lib/synthax/form.rkt
raco pkg install: packages installed, although setup reported errors
```

**Comment by mflatt** [comment]
Is this the problem fixed by #2745 (not included in the release)?

**Comment by pmatos** [comment]
Right, sorry for the noise.
